Describe yellow edge (#555)
VirtusLab/git-machete-intellij-plugin: README.md

### Push

After a rebase, you might want to push the rebased branch to the remote.

> Right-click on the given branch and select `Push...` from a context menu.

Reviewer: While you're here... pls add a mention that selection between target repos is provided by the standard push dialog

Right-click on the given branch and select `Push...` from a context menu.
Push dialog will appear with a proper push button (regular push or force push) depending on which of them is needed.
Basically, you need a force push in a case when you rebased your branch and thus local and remote branch diverged from each other (they have different commits), so git won't let you do a regular push.

> Note that the standard allows for selecting the target remote to push (just like the usual push dialog).

Reviewer: `that the dialog allows`?

### Slide in branch

You can also create a new branch below the selected one (also without editing `machete` file manually).
To do this, choose the `Slide In Branch Below...` action from a context menu.
Type the name of new branch in the dialog window and click `Slide In` button.
Another dialog will appear &mdash; it is a standard new branch IntelliJ dialog.

![](docs/slide_in_new_branch.gif)

A slide in action can also add a branch that already exists in git repository to the branch layout.

> Again, select the `Slide In Branch Bellow...` from a context menu and type the name of an already existing branch.

Reviewer: `Below`

this is what Git Machete displays when [toggle listing commits](#toggle-listing-commits) is enabled.

If one of branch layout edges is yellow, that means the fork point of this branch cannot be unambiguously determined.

> To be precise (and looking at the sample below) the downstream branch (`call-ws`) is still in sync with the upstream (`develop`),

Reviewer: Just:
```suggestion
In the sample below, the downstream branch (`call-ws`) is still in sync with the upstream (`develop`),
```

but the upstream branch tip isn't the fork point of the downstream branch.

Translating from git-ese to English, there are probably commits from some other branches on the way between `develop` and `call-ws`.
You won't come across the yellow edges very often in day-to-day work with git machete

> &mdash; it mostly happens when the tree structure has been surgically modified.

Reviewer: ```suggestion
&mdash; it mostly happens when the branch layout has been surgically modified.
```

However, Git Machete can give you a hint which commit is probably the fork point.
Show commits to see the suggested fork point (see the [toggle listing commits](#toggle-listing-commits) section).

Now you can use `Override Fork Point...` action to choose the fork point of this branch.

> It can be commit inferred by Git Machete (the one marked in commits list), or the one that parent branch is pointing to.

Reviewer: ```suggestion
It can be the commit inferred by Git Machete (the one marked in commits list), or the one that parent branch is pointing to.
```
